src/common/utils/companion.ts

```diff
@@ -0,0 +1,73 @@
+import { PROTOCOL_TOKEN_ADDRESS } from '@common/mocks/tokens';
+import { FullPosition, Permission, Position } from '@types';
+
+export const getCompanionNeedsWithdraw = (position?: Nullable<FullPosition | Position>) => {
+  if (!position) {
+    return false;
+  }
+
+  const toHasYield = position.to.underlyingTokens.length;
+  const toIsProtocol = position.to.address === PROTOCOL_TOKEN_ADDRESS;
```

Comment:
Some other day I'd try to abstract comparing addresses to a utils so you can do the lowercase thingy... (so annoying). Maybe ethers already has something?

Reply:
Cool made this linear ticket to change it later https://linear.app/balmy/issue/BLY-1092/change-all-the-address-comparisons-to-use-the-sdk-one